sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/parquet/ParquetPartitionReaderFactory.scala

```diff
@@ -86,17 +90,45 @@ case class ParquetPartitionReaderFactory(
 
   private val parquetReaderCallback = new ParquetReaderCallback()
 
-  private def getFooter(file: PartitionedFile): ParquetMetadata = {
-    val conf = broadcastedConf.value.value
-    if (aggregation.isDefined || enableVectorizedReader) {
-      // There are two purposes for reading footer with row groups：
-      // 1. When there are aggregates to push down, we get max/min/count from footer statistics.
-      // 2. When there are vectorized reads, we can avoid reading the footer twice by reading
-      //    all row groups in advance and filter row groups according to filters that require
-      //    push down (no need to read the footer metadata again).
-      ParquetFooterReader.readFooter(conf, file, ParquetFooterReader.WITH_ROW_GROUPS)
+  private def openFileAndReadFooter(file: PartitionedFile):
+      (Option[HadoopInputFile], Option[SeekableInputStream], ParquetMetadata) = {
+    val hadoopConf = broadcastedConf.value.value
+    if (aggregation.isDefined) {
+      // When there are aggregates to push down, we get max/min/count from footer statistics.
+      val footer = ParquetFooterReader.readFooter(
+        hadoopConf, file, ParquetFooterReader.WITH_ROW_GROUPS)
+      (None, None, footer)
     } else {
-      ParquetFooterReader.readFooter(conf, file, ParquetFooterReader.SKIP_ROW_GROUPS)
+      // When there are vectorized reads, we can avoid
```

Comment:
Can we extract this into a new util method in `ParquetFooterReader` which perhaps returns the footer and also the input stream? we can then avoid duplicating this code in two places.